utils/datetime.ts

```diff
@@ -18,7 +18,7 @@ const shouldTruncate = new Map<Unit, boolean>([
 export function isWithin(
     a: Date,
     b: Date,
-    timeZone: string | undefined,
+    timeZone: string = new Intl.DateTimeFormat().resolvedOptions().timeZone,
```

Comment:
@haardikdharma10 this default parameter (and the one in `isEqual`) could technically be omitted as it will get picked up by `getDiff`'s, but having in both/all places isn't bad, per se. Up to you if you want to keep or remove.

Reply:
Hey @calebroseland, thanks for your feedback. Got your point but I think let us keep it this way, the code looks much better with the homogeneity? Let me know what you think and if this PR is ready to merge:)

Reply:
Just need a QA review and it should be good to go.